### PR TITLE
fix: update lsp error message of 'relative import path' to 'use deno add' for npm/jsr packages

### PR DESCRIPTION
**Pull request by HasanAlrimawi**
This PR addresses #24033 

**Problem**
LSP shows a message saying that the relative import path isn't prefixed with / or ./ or ../, while the package could be from npm or jsr.

**Fix**
This type of message arises when a Resolution error is received within the `fn to_lsp_diagnostic` in _cli/lsp/diagnostics.rs_.
Then a check is needed to see if the package name starts with @ in order to know if it's local or from npm/jsr.
After that the message can be updated to tell the user to use `deno add` in order to add the dependency.

**Screenshot of the fix**
deno lsp  deno add

### CODE REVIEW

**Comment by HasanAlrimawi** [comment]
Hello @lucacasonato ,
Could you please review this PR when you have time? Thanks in advance.

**Comment by bartlomieju** [comment]
@nayeemrmn PTAL

**Review by nayeemrmn — COMMENTED** [review]
Does this work when there's no import map? Also needs a test.

File: cli/lsp/diagnostics.rs [hunk 1]
```diff
@@ -1272,12 +1274,24 @@ impl DenoDiagnostic {
...
       Self::Redirect { from, to} => (lsp::DiagnosticSeverity::INFORMATION, format!("The import of \"{from}\" was redirected to \"{to}\"."), Some(json!({ "specifier": from, "redirect": to }))),
-      Self::ResolutionError(err) => (
+      Self::ResolutionError(err) => {
+        let mut message;
+        message = enhanced_resolution_error_message(err);
+        if let deno_graph::ResolutionError::ResolverError {error, ..} = err{
+          if let ResolveError::Other(resolve_error, ..) = (*error).as_ref() {
+            if let Some(ImportMapError::UnmappedBareSpecifier(specifier, _)) = resolve_error.downcast_ref::<ImportMapError>() {
+              if specifier.chars().next().unwrap_or('\0') == '@'{
+                message = format!("Use [deno add {}] to add the dependency.", specifier);
```
Comment (nayeemrmn) [inline cli/lsp/diagnostics.rs:1284]: I think the hint should be appended to the original message here.

**Pull request merged**